### PR TITLE
build: refactored code

### LINKED ISSUE #482: Add build badges and update test coverage config
What should be done:
1. Add CI build badge at the top of the https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md with the current CI build status (success or fail)
2. Add test coverage badge with % of lines covered by tests at the top of the https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md
3. Remove the following files from the test coverage report:

- `src/dollar/libs/*`
- `src/dollar/mocks/*`
- `src/dollar/utils/*`
- `src/manager/libraries/*`
- `src/ubiquistick/MockUBQmanager.sol`
- `test/*`

Build badge docs: https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

### PR DESCRIPTION
**Pull request by FibrinLab**
Resolves #482

### CODE REVIEW

**Comment by FibrinLab** [comment]
@pavlovcik I noticed there were conflicting commits. I refactored the code and gave coverage its own CI

File: README.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+![CodeQL](https://github.com/ubiquity/ubiquity-dollar/actions/workflows/codeql-analysis.yml/badge.svg?branch=development)
+![Conventional Commits](https://github.com/ubiquity/ubiquity-dollar/actions/workflows/conventional-commits.yml/badge.svg?branch=development)
+![Build and Test](https://github.com/ubiquity/ubiquity-dollar/actions/workflows/build-and-test.yml/badge.svg?branch=development)
+![Yarn Audit](https://github.com/ubiquity/ubiquity-dollar/actions/workflows/yarn-audit.yml/badge.svg?branch=development)
+![Slither Analysis](https://github.com/ubiquity/ubiquity-dollar/actions/workflows/slither-analysis.yml/badge.svg?branch=development)
+
+![](https://byob.yarr.is/ubiquity/ubiquity-dollar/coverage)
```
Comment (0x4007) [inline README.md]: ?

Comment (FibrinLab) [reply README.md]: Thats for the build badges. You requested I update README to show all workflows

Comment (0x4007) [reply README.md]: You made line 7 not render

**Review by 0x4007 — CHANGES_REQUESTED** [review]
Does not resolve bounty spec:

> Add test coverage badge with % of lines covered by tests at the top of the https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md

File: packages/contracts/package.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
     "clean:yarn": "yarn clean",
     "clean:forge": "forge clean",
     "build": "forge build",
+    "coverage": "forge coverage --fork-url https://eth.ubq.fi/v1/mainnet",
```
Comment (0x4007) [inline packages/contracts/package.json]:
```suggestion
    "test:coverage": "forge coverage --fork-url https://eth.ubq.fi/v1/mainnet",
```

Comment (FibrinLab) [reply packages/contracts/package.json]: Done

**Comment by FibrinLab** [comment]
@pavlovcik 
"${{ secrets.GITHUB_TOKEN }} doesn't seem to work. 

To get coverage to work you would need to create a personal access token and Select "repo scope", then add to the ubiquity-dollar secrets variable "ACTIONS_TOKEN". The re-run the action.

It works fine.

**Comment by FibrinLab** [comment]
Screenshot 2023-02-17 at 22 12 50

@pavlovcik it needs access token for permissions with repo scope to output to ReadMe correctly

**Comment by FibrinLab** [comment]
@pavlovcik Is it possible to update GITHUB_TOKEN permissions to include repo creation?

**Pull request closed without merging**